.github/workflows/docs.yaml

```diff
@@ -0,0 +1,20 @@
+name: Build Documentation
+
+on:
+  pull_request:
+    branches: [ main ]
+    paths:
+      - 'docs/**'
+
+jobs:
+
+  build-docs:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Build docs and check for errors
+        run: bash scripts/build_docs.sh
```

Comment:
moving this to its own file so that we can trigger this to run only when the docs change. this way we don't run this job when we simply update python code

Reply:
Nice optimization